lib/sandthorn_driver_sequel.rb

```diff
@@ -3,20 +3,81 @@
 require "sandthorn_driver_sequel/wrappers"
 require "sandthorn_driver_sequel/event_query"
 require "sandthorn_driver_sequel/event_store_context"
-require "sandthorn_driver_sequel/access"
-require "sandthorn_driver_sequel/storage"
 require 'sandthorn_driver_sequel/event_store'
 require 'sandthorn_driver_sequel/errors'
 require 'sandthorn_driver_sequel/migration'
+require 'yaml'
 
 module SandthornDriverSequel
   class << self
-    def driver_from_url url: nil, context: nil
-      EventStore.new url: url, context: context
-    end
+    
     def migrate_db url: nil, context: nil
       migrator = Migration.new url: url, context: context
       migrator.migrate!
     end
+
+    def driver_from_url url: nil, context: nil
+      yield(session_configure) if block_given?
+      driver = EventStore.new url: url, context: context, event_serializer: configuration.event_serializer, event_deserializer: configuration.event_deserializer
+      @session_configuration = nil
+      return driver
+    end
+
+    def driver_from_connection connection: nil, context: nil
+      yield(session_configure) if block_given?
+      driver = EventStore.new connection: connection, context: context, event_serializer: configuration.event_serializer, event_deserializer: configuration.event_deserializer
+      @session_configuration = nil
+      return driver
+    end
+
+    def configure
+      yield(configuration) if block_given?
+    end
+
+    private
+
+    def session_configure
```

Comment:
If a session configuration is used to avoid global config, why is this memoized?